flowmachine/flowmachine/core/query_state.py

```diff
@@ -0,0 +1,346 @@
+        state, trigger_success = self.state_machine.trigger(event)
+        return QueryState(state.decode()), trigger_success
+
+    def cancel(self):
+        """
+        Attempt to mark the query as cancelled.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            triggered the query to be cancelled with this call
+
+        """
+        return self.trigger_event(QueryEvent.CANCEL)
+
+    def enqueue(self):
+        """
+        Attempt to mark the query as queued.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            triggered the query to be queued with this call
+
+        """
+        return self.trigger_event(QueryEvent.QUEUE)
+
+    def raise_error(self):
+        """
+        Attempt to mark the query as having errored while running.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            marked the query as erroring with this call
+
+        """
+        return self.trigger_event(QueryEvent.ERROR)
+
+    def execute(self):
+        """
+        Attempt to mark the query as in the process of executing.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            marked the query as executing with this call
+
+        """
+        return self.trigger_event(QueryEvent.EXECUTE)
+
+    def finish(self):
+        """
+        Attempt to mark the query as completed.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            marked the query as finished with this call
+
+        """
+        return self.trigger_event(QueryEvent.FINISH)
+
+    def reset(self):
+        """
+        Attempt to mark the query as in the process of resetting.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            triggered the query to be reset with this call
+
+        """
+        return self.trigger_event(QueryEvent.RESET)
+
+    def finish_resetting(self):
+        """
+        Attempt to mark the query as having finished resetting.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+           marked the reset as complete with this call.
+
+        """
+        return self.trigger_event(QueryEvent.FINISH_RESET)
+
+    def wait_until_complete(self):
+        """
+        Blocks while the query is in any state that makes how to get
```

Comment:
This docstring is a little difficult to understand. How about rephrasing it as follows?
```
Blocks until the query is in a state where its result is determinate (i.e., one of "know", "errored", "completed", "cancelled").
```